### Preparing data for analysis

In [1]:
# library to easily deal with data
import pandas as pd

In [2]:
df = pd.read_csv('data.csv') # importing the data
df.head(2)

Athlete  Status                Hometown   Age  Height  Weight  \
0      aalon-cruz  Active  Parrish, United States  30.0    72.0   145.0   
1  aaron-phillips  Active  Houston, United States  30.0    69.0   135.0   

   Octagon Debut  Reach  Leg reach  Sig.str by target : HEAD  ...  \
0  Jul. 30, 2019   78.0       42.0                       0.0  ...   
1  May. 24, 2014   70.0       41.0                      28.0  ...   

   Sig. Strikes Attempted  Takedowns Landed Takedowns Attempted  \
0                     NaN               NaN                 NaN   
1                    94.0               NaN                 1.0   

  Striking accuracy Grappling accuracy Wins by Decision Trains at  \
0               NaN                NaN              NaN       NaN   
1               60%                 0%              NaN       NaN   

  First Round Finishes Fighting style  Title Defenses  
0                  NaN            NaN             NaN  
1                  NaN            NaN             NaN  

[2 rows x 40 columns]

In [3]:
df = df.set_index('Athlete') # reindexing the dataframe with the list of athletes
df.head(2)

Status                Hometown   Age  Height  Weight  \
Athlete                                                                
aalon-cruz      Active  Parrish, United States  30.0    72.0   145.0   
aaron-phillips  Active  Houston, United States  30.0    69.0   135.0   

                Octagon Debut  Reach  Leg reach  Sig.str by target : HEAD  \
Athlete                                                                     
aalon-cruz      Jul. 30, 2019   78.0       42.0                       0.0   
aaron-phillips  May. 24, 2014   70.0       41.0                      28.0   

                Sig.str by target : BODY  ...  Sig. Strikes Attempted  \
Athlete                                   ...                           
aalon-cruz                           0.0  ...                     NaN   
aaron-phillips                      24.0  ...                    94.0   

               Takedowns Landed Takedowns Attempted Striking accuracy  \
Athlete                                                                 
aalon-cruz                  NaN                 NaN               NaN   
aaron-phillips              NaN                 1.0               60%   

               Grappling accuracy Wins by Decision Trains at  \
Athlete                                                        
aalon-cruz                    NaN              NaN       NaN   
aaron-phillips                 0%              NaN       NaN   

               First Round Finishes  Fighting style  Title Defenses  
Athlete                                                              
aalon-cruz                      NaN             NaN             NaN  
aaron-phillips                  NaN             NaN             NaN  

[2 rows x 39 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 960 entries, aalon-cruz to nikolas-motta-0
Data columns (total 39 columns):
Status                        938 non-null object
Hometown                      920 non-null object
Age                           953 non-null float64
Height                        912 non-null float64
Weight                        903 non-null float64
Octagon Debut                 960 non-null object
Reach                         873 non-null float64
Leg reach                     836 non-null float64
Sig.str by target : HEAD      893 non-null float64
Sig.str by target : BODY      893 non-null float64
Sig.str by target : LEG       893 non-null float64
Sig.str : Standing            893 non-null object
Sig.str : Clinch              893 non-null object
Sig.str : Ground              893 non-null object
wins by : KO/TKO %            893 non-null object
wins by : Decision %          893 non-null object
wins by Submission %          893 non-null object
Sig. Str. Landed per 

First of all, The dataset contains 960 rows and about 39 data columns. Each row represents a fighter's stats, hometown, reach,...

the columns 'wins by : KO/TKO %', 'wins by : Decision %', 'wins by Submission %' and 'Wins by Knockout','Wins by Submission', 'Wins by Decision' contain the same information. The same thing with 'Striking accuracy', 'Grappling accuracy' and 'Sig. Strikes Landed', 'Sig. Strikes Attempted', 'Takedowns Landed', 'Takedowns Attempted'. Let's drop these columns : 

In [5]:
df.drop(columns={'wins by : KO/TKO %', 'wins by : Decision %', 'wins by Submission %',
                'Striking accuracy', 'Grappling accuracy'}, axis=1, inplace=True)

some columns have decimal values instead of integer values, let's fix that :

In [6]:
# Converting columns type from float to int

df[['Age', 'Sig.str by target : HEAD', 'Sig.str by target : BODY', 'Sig.str by target : LEG',
    'Fight Win Streak', 'Wins by Knockout', 'Wins by Submission', 'Sig. Str. Absorbed per win',
        'Sig. Str. Defense', 'Takedown Defense', 'Sig. Strikes Landed', 'Sig. Strikes Attempted',
            'Takedowns Landed', 'Takedowns Attempted', 'Wins by Decision', 'First Round Finishes',
                 'Title Defenses']] = df[['Age', 'Sig.str by target : HEAD', 'Sig.str by target : BODY',
    'Sig.str by target : LEG', 'Fight Win Streak', 'Wins by Knockout',
        'Wins by Submission', 'Sig. Str. Absorbed per win', 'Sig. Str. Defense',
            'Takedown Defense', 'Sig. Strikes Landed', 'Sig. Strikes Attempted',
                'Takedowns Landed','Takedowns Attempted', 'Wins by Decision', 'First Round Finishes', 'Title Defenses']].fillna(0).astype(int)

In [7]:
# checking other columns
df[['Sig.str : Standing', 'Sig.str : Clinch', 'Sig.str : Ground']].head(2)

Sig.str : Standing Sig.str : Clinch Sig.str : Ground
Athlete                                                            
aalon-cruz                0 (0 %)         0 (0 %)           0 (0 %)
aaron-phillips           23 (41%)        20 (36%)          13 (23%)

we need to get rid of the % + converting the columns type to integer :

In [8]:
df['Sig.str : Standing'] = df['Sig.str : Standing'].fillna('0').apply(lambda x : x.split(' ')[0]).astype(int)
df['Sig.str : Clinch'] = df['Sig.str : Clinch'].fillna('0').apply(lambda x : x.split(' ')[0]).astype(int)
df['Sig.str : Ground'] = df['Sig.str : Ground'].fillna('0').apply(lambda x : x.split(' ')[0]).astype(int)

In [9]:
df.dtypes

Status                         object
Hometown                       object
Age                             int32
Height                        float64
Weight                        float64
Octagon Debut                  object
Reach                         float64
Leg reach                     float64
Sig.str by target : HEAD        int32
Sig.str by target : BODY        int32
Sig.str by target : LEG         int32
Sig.str : Standing              int32
Sig.str : Clinch                int32
Sig.str : Ground                int32
Sig. Str. Landed per win       object
Fight Win Streak                int32
Wins by Knockout                int32
Wins by Submission              int32
Sig. Str. Absorbed per win      int32
Takedown avg per 15 min       float64
Submission avg per 15 min     float64
Sig. Str. Defense               int32
Takedown Defense                int32
Knockdown Ratio                object
Average fight time             object
Sig. Strikes Landed             int32
Sig. Strikes

In [10]:
df.to_csv('prep_data.csv') # save the dataset to a csv file